src/classes/tmDaoProvisioning.cls

```diff
@@ -0,0 +1,174 @@
+        return countryRevMap.Values();
+    }
+
+    public static String getCurrencyCode(String countryIsoCode) {
+        String currencyCode = tmConstants.TFR_DEFAULT_CURRENCY;
+        Map<String, sfbase__CountryRevenueMap__c> countryRevMap = getCountryRevMap();
+        if(countryIsoCode != null) {
+            if(countryRevMap.size() > 0 && countryRevMap.get(countryIsoCode) != null) {
+                currencyCode = countryRevMap.get(countryIsoCode).sfbase__PrimaryCurrency__c;
+            }
+        } 
+        return currencyCode;
+    }
+
+    public static String getPaymentTerm(String countryIsoCode) {
+        String paymentTerm = tmConstants.CONTRACT_PAYMENT_TERM;
+        sfbase__CountryRevenueMap__c revMapping = getcountryRevMap().get(countryIsoCode);
+        if (revMapping != null) {
+            String revOwner = revMapping.sfbase__RevenueOwner__c;        
+
+            List<sfbase__SOBPaymentMethodMap__c> paymentMethodMapping = [SELECT sfbase__DefaultPaymentTerm__c
+                                                        FROM sfbase__SOBPaymentMethodMap__c
+                                                        WHERE sfbase__PaymentMethod__c = :tmConstants.CONTRACT_PAYMENT_METHOD
+                                                        AND sfbase__RevenueOwner__c = :revOwner
+                                                        LIMIT 1];
+            if (paymentMethodMapping.size() > 0) {
+                paymentTerm = paymentMethodMapping.get(0).sfbase__DefaultPaymentTerm__c;
+            }                                                   
+        }
+        return paymentTerm;
+    }  
+
+    public static Id createOrderOnNewContract(Contract c, Order ord, List<OrderItem> ordItems, Id priceBookId) {
+        return provisioningHelper.doCreateOrderOnNewContractWithoutSharing(c, ord, ordItems, priceBookId);
+    }
+
+    public static Id createOrderOnExistingContract(Id contractId, Order ord, List<OrderItem> ordItems) {
+        return provisioningHelper.doCreateOrderOnNewContractWithoutSharing(contractId, ord, ordItems);
+    }
+    public static id upgradeOrderOnExistingContract(Id contractId, Order ord, List<OrderItem> ordItems, List<OrderItem> oldOrdItems) {
+        provisioningHelper.updateOrderItems(oldOrdItems);
+        return provisioningHelper.doCreateOrderOnNewContractWithoutSharing(contractId, ord, ordItems);
+    }
+    private static tmDaoProvisioningHelper provisioningHelper {
+        get {
+            if (provisioningHelper == null) {
+                provisioningHelper = new tmDaoProvisioningHelper();
+            }
+            return provisioningHelper;
+        }
+        private set;
+    }
+
+    private without sharing class tmDaoProvisioningHelper {
+        public Id doCreateOrderOnNewContractWithoutSharing(Contract c, Order ord, List<OrderItem> ordItems, Id pbId) {
+            c.RecordTypeId = recordTypeIds.get('Contract');
+            insert c;
```

Comment:
This DML statement is not surrounded by any try-catch block. Please add a try-catch block to ensure exceptions are handled properly.